apps/server-asset-sg/src/features/asset-edit/asset-edit.repo.ts

```diff
@@ -49,6 +49,7 @@ export class AssetEditRepo implements Repo<AssetEditDetail, number, AssetEditDat
   }
 
   async create(data: AssetEditData): Promise<AssetEditDetail> {
+    await this.validateReferencesOrThrow(data);
```

Comment:
The repo should not be responsible for validating its data - move this to either a service or to a controller.